### PR TITLE
cli: clarify `jj branch forget` help

### PR DESCRIPTION
**Pull request by martinvonz**
See how this looks, @jsoref. I'm happy to update it if it's still unclear.

### CODE REVIEW

File: src/commands.rs [hunk 1]
```diff
@@ -697,8 +697,11 @@ enum BranchSubcommand {
         names: Vec<String>,
     },
 
-    /// Delete the local version of an existing branch, without propagating the
-    /// deletion to remotes.
+    /// Forget everything about a branch, including its local and remote
+    /// targets.
+    ///
+    /// A forgotten branch will not impact remotes on future pushes. It will be
+    /// recreated on future pulls if it still exists on the remote.
```
Comment (jsoref) [inline src/commands.rs:704]: My personal preference would be:
```suggestion
    /// recreated on future pulls if it still exists in the remote.
```

Comment (martinvonz) [reply src/commands.rs]: SGTM. Updated.

Comment (arxanas) [reply src/commands.rs]: Do we generally say e.g. "in the remote machine" rather than "on the remote machine"? The suggestion sounds unusual to me.

Comment (martinvonz) [reply src/commands.rs]: Prepositions are hard. I'm not even native English speaker, but I'm not sure how much that matters anyway - prepositions are pretty arbitrary in most languages, I think. Both versions sound fine to me. FWIW, `git help remote`, `git help push`, and `git help pull` seem to prefer "on the remote", but I haven't really analyzed the contexts in which they're using them.

Comment (jsoref) [reply src/commands.rs]: So, I clearly don't think of/read `remote` as a machine. I see it as a container (hence `in`).

If it had said `remote machine`, I wouldn't have made the suggestion.

Comment (jsoref) [reply src/commands.rs]: Fwiw, `git help remote` describes remote as a repository and not a machine. I would definitely use `in` for a repository and not `on`.

Comment (jsoref) [reply src/commands.rs]: Fwiw, `git help remote` half agrees with me:

```
       git-remote - Manage set of tracked repositories
```

```
           Fetch updates for remotes or remote groups in the repository as
```

```
           With --tags option, git fetch <name> imports every tag from the
           remote repository.
```

Note that this case is actually different, `performing an action on` a thing refers to targeting the thing which is a special case:

```
       are available to perform operations on the remotes.
```

But, yes, prepositions are incredibly quirky language to language and speaker to speaker.

Comment (martinvonz) [reply src/commands.rs]: I'm fine with either, and you seem to have a stronger opinion about this than @arxanas does so I'm leaning towards "in the remote", but I'll give @arxanas a chance to comment again before I resolve this.

@jsoref, does the PR look good to you now? Please review and approve it if it does (under the "Files changed" tab).

**Review by jsoref — APPROVED** [review]

File: src/commands.rs (hunk 1 above)
Comment (jsoref) [reply src/commands.rs]: Yep. ✅

**Pull request merged**